server/apps/a12n/views.py

```diff
@@ -0,0 +1,163 @@
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework_simplejwt.tokens import RefreshToken
+
+from .serializers import UserRegisterSerializer, UserLoginSerializer
+from .models import UserSession
+from apps.account.models import User
+
+from django.contrib.auth.tokens import default_token_generator
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from django.utils.encoding import force_bytes, force_str
+from django.core.mail import BadHeaderError, send_mail
+
+from datetime import timezone
+
+
+class RegistrationAPIView(APIView):
+    """
+    Представление для регистрации пользователя.
+
+    Методы:
+        - POST: Регистрирует пользователя и отправляет письмо с
+        подтверждением регистрации.
+    """
+    def post(self, request):
+        serializer = UserRegisterSerializer(data=request.data)
+        if serializer.is_valid():
+            user = serializer.save(is_active=False)
+
+            token = default_token_generator.make_token(user)
+            uid = urlsafe_base64_encode(force_bytes(user.pk))
+
+            confirm_url = f'/confirm/{uid}/{token}'
+            
+            try:
+                send_mail(
+                    'Register confirm',
+                    f'Hi, {user.username}, please confirm your registration,
+                    follow this link: {confirm_url}',
+                    'example@example.com',
+                    [user.email],
+                )
+            except BadHeaderError:
+                return Response(
+                    {'detail': 'Error sending confirmation. Try again.'},
+                    status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                )
+            
+        return Response(
+            serializer.errors,
+            status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
If u will test, u will receive a 400 error regardless of sending the email.
Try to solve it yourself.